### ISSUE TITLE
Local file reference

### ISSUE THREAD

**Issue by maxsalven** [jonasschmidt/ex_json_schema#12]
How can I parse a schema with a local file reference (`common.json#/example`)?

```
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "Example Schema",
  "type": "object",
  "required": [ "example" ],
  "additionalProperties": false,
  "properties": {
    "example": { "$ref": "common.json#/example" }
  }
}
```

If I just do:
`File.read!("lib/myapp/schemas/myschema.json") |> Poison.decode! |> ExJsonSchema.Schema.resolve`
then we get a (confusing) error:
`** (BadFunctionError) expected a function, got: nil`.

Understandably the resolver doesn't know where to look for the file, is there a way I can pass in a base directory or similar?

**Comment by jonasschmidt** [comment]
You can configure your own resolver function like this:
```elixir
config :ex_json_schema,
  :remote_schema_resolver,
  fn url -> HTTPoison.get!(url).body |> Poison.decode! end
```
It's just a function that gets called with the URL (in your case `common.json`) and has to return a Map. You can make that function look up the file in your local file system.

**Comment by wanderanimrod** [comment]
This has just happened to me and I spent a significant amount of time figuring out what was wrong. Can we throw a more descriptive error when one tries to use `$ref` without defining the `:remote_schema_resolver` function in the config?

I can submit a pull request if this is reasonable.

**Comment by jonasschmidt** [comment]
You're absolutely right. The straight-forward solution would probably be to have the default for the `:remote_schema_resolver` config be a function that just raises an error. If you want to submit a pull request, feel free to do that. If you don't have the time, I will most likely get a chance to fix it in the next few days.

**Comment by jonasschmidt** [comment]
Fixed in version `0.5.3`. Thanks for reporting this!
